File: src/main/java/io/github/aphelium/modules/brokers/dtos/CreateBrokerDTO.java
package io.github.aphelium.modules.brokers.dtos;

import lombok.Data;

import javax.validation.constraints.NotBlank;

@Data
public class CreateBrokerDTO {

    @NotBlank(message = "The name is mandatory")
    private String name;
    @NotBlank(message = "The description is mandatory")
    private String description;
    @NotBlank(message = "The slug is mandatory")
    private String slug;
    @NotBlank(message = "The logo is mandatory")
    private String logo;
    @NotBlank(message = "The date launched is mandatory")
    private String dateLaunched;

}

File: src/main/kotlin/io/github/aphelium/modules/brokers/dtos/CreateBrokerDTO.kt
package io.github.aphelium.modules.brokers.dtos

import javax.validation.constraints.NotBlank

class CreateBrokerDTO {
    var name: @NotBlank(message = "The name is mandatory") String? = null
    var description: @NotBlank(message = "The description is mandatory") String? = null
    var slug: @NotBlank(message = "The slug is mandatory") String? = null
    var logo: @NotBlank(message = "The logo is mandatory") String? = null
    var dateLaunched: @NotBlank(message = "The date launched is mandatory") String? = null
}